### PR TITLE
Adopt Artifacts.test(localEngine:) in a few tests

### PR DESCRIPTION
**Pull request by jmagman**
Replace `MockLocalEngineArtifacts` with `Artifacts.test(localEngine:)` in a few places.

Follow up to #76277
Part of #71511

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart [hunk 1]
```diff
@@ -687,41 +680,7 @@ Information about project "Runner":
       expect(buildPhaseScriptContents.contains('ARCHS=armv7'), isTrue);
     });
 
-    testUsingOsxContext('sets ARCHS=x86_64 when sim local engine is set', () async {
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart:690]: I moved this one into a new `sim local engine` group so the `localArtifacts` could be shadowed outside the test to be overridden in the context.

File: packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart [hunk 2]
```diff
@@ -769,26 +722,67 @@ Information about project "Runner":
       expect(buildPhaseScriptContents.contains('TRACK_WIDGET_CREATION=true'), isFalse);
     });
 
-    testUsingOsxContext('sets ARCHS=armv7 when armv7 local engine is set', () async {
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart:772]: Same.  Moved into `armv7 local engine` group.

**Review by jonahwilliams — APPROVED** [review]
LGTM (but not to CI)

**Comment by jonahwilliams** [comment]
/tmp/flutter sdk/packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart:980: trailing blank line

File: packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart [hunk 3]
```diff
@@ -991,4 +978,3 @@ flutter:
   });
 }
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart:980]:
```suggestion
```

**Pull request merged**